core/discovery/discovery.go

```diff
@@ -44,6 +45,45 @@ const (
 
 const logPrefix = "[discovery] "
 
+// Discovery structure holds discovery service state
+type Discovery struct {
+	identityRegistry            identity_registry.IdentityRegistry
+	ownIdentity                 identity.Identity
+	identityRegistration        identity_registry.RegistrationDataProvider
+	proposalRegistry            ProposalRegistry
+	signerCreate                identity.SignerFactory
+	signer                      identity.Signer
+	proposal                    market.ServiceProposal
+	statusChan                  chan Status
+	status                      Status
+	proposalAnnouncementStopped *sync.WaitGroup
+	unsubscribe                 func()
+	stop                        func()
+
+	sync.RWMutex
+}
+
+// NewService creates new discovery service
+func NewService(
+	identityRegistry identity_registry.IdentityRegistry,
+	identityRegistration identity_registry.RegistrationDataProvider,
+	proposalRegistry ProposalRegistry,
+	signerCreate identity.SignerFactory,
+) *Discovery {
+	return &Discovery{
+		identityRegistry:            identityRegistry,
+		identityRegistration:        identityRegistration,
+		proposalRegistry:            proposalRegistry,
+		signerCreate:                signerCreate,
+		statusChan:                  make(chan Status),
+		status:                      StatusUndefined,
+		proposalAnnouncementStopped: &sync.WaitGroup{},
+		unsubscribe:                 func() {},
```

Comment:
Why we need this useless functions `unsubscribe` and `stop`?
If it's not used now, we should not add it.

Reply:
Existing code, not my scope.
And yes, it's being used